src/telit_he910.cpp

```diff
@@ -0,0 +1,2102 @@
+					{
+						subState = 0;
+						state = POWER_OFF;
+						break;
+					}
+					
+					subState = 0;
+					state = WAIT_FOR_NETWORK;
+					
+					break;
+			}
+		
+			break;
+			
+		case WAIT_FOR_NETWORK:
+		
+			switch(subState)
+			{
+				case 0:
+			
+					timeout = uptimeMs() + NETWORK_CONNECT_TIMEOUT;
+					subState = 1;
+			
+					break;
+			
+				case 1:
+					
+					if(getNetworkConnectionStatus(&connectStatus))
+					{
+						if(connectStatus == REGISTERED_HOME || (device->config.networkOperatorSettings.allowDataRoaming && connectStatus == REGISTERED_ROAMING))
+						{
+							if(getCurrentNetwork(&current_network))
+							{
+								debug("Telit connected to PLMN %u, access type %u", current_network.PLMN, current_network.networkType);
+							}
+							subState = 0;
+							state = CLOSE_PDP;
+						}
+						else
+						{
+							timer = uptimeMs() + 500;
+							subState = 2;
+						}
+					}
+					
+					break;
+					
+				case 2:
+				
+					if(uptimeMs() > timeout)
+					{
+						subState = 0;
+						state = POWER_OFF;
+					}
+					else if(uptimeMs() > timer)
+					{
+						subState = 1;
+					}
+				
+					break;
+			}
+		
+			break;
+			
+		case CLOSE_PDP:
+		
+			// deactivate data session (just in case the network thinks we still have an active PDP context)
+			if(closePDPContext())
+			{
+				subState = 0;
+				state = OPEN_PDP_DELAY;
+			}
+			else
+			{	
+				subState = 0;
+				state = POWER_OFF;
+			}
+
+			break;
+			
+		case OPEN_PDP_DELAY:
+		
+			switch(subState)
+			{
+				case 0:
+				
+					timer = uptimeMs() + 1000;
+					subState = 1;
+					
+					break;
+					
+				case 1:
+
+					if(uptimeMs() > timer)
+					{
+						state = OPEN_PDP;
+						subState = 0;
+					}
+					
+					break;
+			}
+			
+			break;
+			
+		case OPEN_PDP:
+		
+			// activate data session
+			if(openPDPContext())
+			{
+				pdp_counter = 0;
+				state = READY;
+			}
+			else
+			{	
+				if(pdp_counter < PDP_MAX_ATTEMPTS)
+				{
+					pdp_counter++;
+					state = CLOSE_PDP;
+				}
+				else
+				{
+					pdp_counter = 0;
+					state = POWER_OFF;
+				}
+			}
+		
+			break;
+			
+		case READY:
+		
+			switch(subState)
+			{
+				case 0:
+				
+					if(getNetworkConnectionStatus(&connectStatus), (connectStatus != REGISTERED_HOME && connectStatus != REGISTERED_ROAMING))
+					{
+						debug("Modem has lost network connection");
+						connect = false;
+						state = WAIT_FOR_NETWORK;
+					}
+					else if(getPDPContext(&pdp_connected), !pdp_connected)
+					{
+						debug("Modem has lost data session");
+						connect = false;
+						state = CLOSE_PDP;
+					}
+					else
+					{
+						connect = true;
+						timer = uptimeMs() + 1000;
+						subState = 1;
+					}
+				
+					break;
+					
+				case 1:
+				
+					if(uptimeMs() > timer)
+						subState = 0;
+				
+					break;
+			}
+		
+			break;
+			
+		default:
+		
+			state = POWER_OFF;
+		
+			break;
+	}
+	
+	return state;
+}
+
+void openxc::telitHE910::deinitialize() {
```

Comment:
This brace style is right, but elsewhere in this file it's not matching the code base.